addons/website_form_project/controllers/website_form.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website_form.controllers import main
+from odoo import tools
+
+class WebsiteForm(main.WebsiteForm):
+    def insert_record(self, request, model, values, custom, meta=None):
+        if model.model == 'project.task' and 'email_from' in values:
+            values_email_normalized = tools.email_normalize(values.get('email_from'))
+            visitor_sudo = request.env['website.visitor']._get_visitor_from_request()
+            visitor_partner = visitor_sudo.partner_id
+
+            if values_email_normalized and not visitor_partner
```

Comment:
This is literally a copy paste from website_crm, can't this be done inside the `insert_record` of `website_form` module?

Something like "if the model has a partner_id field, then do that code" -> and remove the 2 occurences in website_form_project and website_crm added here

Reply:
I was thinking something along these lines, or at least create a helper function in `website_form` with the field names as arguments.